marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -718,6 +775,7 @@ instance caseHasContractData :: HasContractData Case where
 data Value
   = AvailableMoney AccountId (Term Token)
   | Constant BigInteger
+  | ConstantParam String
```

Comment:
Instead of ConstantParam, shouldn't this be TemplateVariable? or ConstantFromTemplate?

Reply:
I don't think `TemplateVariable` because I like this idea of strongly typed template variables. Would `ValueParam` be better?

Reply:
Ahem, I asked about this in #marlowe-team and only Simon responded ¬¬
Anyway, maybe `ValueParam` would work, but it is going to be a constant, not an arbitrary `Value` so I thought it was more specific to call it constant. The problem I see with calling it `Template` is that, a template would be the whole contract, not the construct in particular, that would be more like a "hole" in the template, I think.

Reply:
I saw the message in the slack but didn't know how to respond as I didn't had the other names in my head 😅. Not sure about @shmish111 concern about strongly typed template variables. Was the fact that we were missing `Constant` as part of `TemplateVariable` the problem?.

My current concern is that having `Constant BigInteger` and `ConstantParam String` lets me think that we have two constant types, one of type BigInteger and one of type String. Maybe having `ParameterizedConstant String` would be clearer in the intent that it is still a constant, but it is coming from a parameter somewhere. 

Or maybe just having `ConstantParam TemplateVariableIdentifier`, where the later is just an alias to String

Reply:
my concern is basically the same as what @palas  said, template suggests a sort of `Any` type of template, just like holes are, where as we want a template type that is specific to values and another that is specific to timeouts. I now agree with @palas that `ConstantParam` is correct.

Reply:
And how about the type `ConstantParam TemplateVariableIdentifier` with `TemplateVariableIdentifier` being a type alias to String?

Reply:
I don't see any value in `TemplateVariableIdentifier` personally

Reply:
at least a comment in the constructor? 😅

Reply:
Sure, but I really don't wanna run the CI again just for that :)
We don't have any comments in the semantics types yet, but they are the best documented bit, so I doubt that would be a source of confusion. We used to have a comment like `--     ^ numerator     ^ denominator` but they always got unaligned